File: langs/StdLibs/one.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text.RegularExpressions;

public class OneRegex
{
    public static string[] MatchFromIndex(string pattern, string input, int offset)
    {
        var match = new Regex($"\\G{pattern}").Match(input, offset);
        if (!match.Success) return null;
        var result = match.Groups.Cast<Group>().Select(g => g.Value).ToArray();
        return result;
    }
}

public class OneReflect
{
    public static Dictionary<string, OneClass> PublishedTypes = new Dictionary<string, OneClass>(StringComparer.OrdinalIgnoreCase);

    public static OneClass GetClass(object obj) => GetClassByName(obj.GetType().Name);
    public static OneClass GetClassByName(string name) => PublishedTypes.GetValueOrDefault(name);
    public static void Publish(Type type) => PublishedTypes[type.Name] = new OneClass(type);
}

public class OneClass
{
    public Type Native { get; set; }

    public string Name => Native.Name;
    public static Dictionary<string, OneField> Fields;
    public static Dictionary<string, OneMethod> Methods;

    public OneClass(Type native)
    {
        Native = native;
        Fields = native.GetFields().Select(x => new OneField(x)).ToDictionary(x => x.Name, StringComparer.OrdinalIgnoreCase);
        Methods = native.GetMethods().Select(x => new OneMethod(x)).ToDictionary(x => x.Name, StringComparer.OrdinalIgnoreCase);
    }

    public OneField GetField(string name) => Fields.GetValueOrDefault(name);
    public OneMethod GetMethod(string name) => Methods.GetValueOrDefault(name);
    public List<OneField> GetFields() => Fields.Values.ToList();
    public List<OneMethod> GetMethods() => Methods.Values.ToList();
}

public class OneField
{
    public FieldInfo Native { get; set; }

    public string Name => Native.Name;
    public bool IsStatic => Native.IsStatic;

    public OneField(FieldInfo native)
    {
        Native = native;
    }

    public object GetValue(object obj) => Native.GetValue(obj);
    public void SetValue(object obj, object value) => Native.SetValue(obj, value);
}

public class OneMethod
{
    public MethodInfo Native { get; set; }

    public string Name => Native.Name;
    public bool IsStatic => Native.IsStatic;

    public OneMethod(MethodInfo native)
    {
        Native = native;
    }

    public object Call(object obj, object[] args) => Native.Invoke(obj, args);
}

File: langs/StdLibs/one.java
import java.util.List;
import java.util.ArrayList;
import java.util.Map;
import java.util.TreeMap;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.lang.reflect.*;

class OneRegex
{
    public static List<String> matchFromIndex(String pattern, String input, int offset) {
        Pattern patternObj = Pattern.compile("\\G" + pattern);
        Matcher matcher = patternObj.matcher(input);
        if (!matcher.find(offset))
            return null;

        List<String> result = new ArrayList();
        result.add(matcher.group());
        for (int i = 0; i < matcher.groupCount(); i++)
            result.add(matcher.group(i));
        return result;
    }
}

class OneReflect
{
    public static Map<String, OneClass> publishedTypes = new TreeMap<String, OneClass>(String.CASE_INSENSITIVE_ORDER);

    public static OneClass getClass(Object obj) {
        return getClassByName(obj.getClass().getName());
    }

    public static OneClass getClassByName(String name) {
        return publishedTypes.get(name);
    }
    
    public static void publish(Class clazz) {
        publishedTypes.put(clazz.getName(), new OneClass(clazz));
    }
}

class OneClass
{
    public Class clazz;
    public String name;

    public static Map<String, OneField> fields = new TreeMap<String, OneField>(String.CASE_INSENSITIVE_ORDER);
    public static Map<String, OneMethod> methods = new TreeMap<String, OneMethod>(String.CASE_INSENSITIVE_ORDER);

    public OneClass(Class clazz)
    {
        this.clazz = clazz;
        this.name = clazz.getName();

        for (Field field : clazz.getDeclaredFields()) {
            OneField oneField = new OneField(field);
            this.fields.put(oneField.name, oneField);
        } 

        for (Method method : clazz.getDeclaredMethods()) {
            OneMethod oneMethod = new OneMethod(method);
            this.methods.put(oneMethod.name, oneMethod);
        } 
    }

    public OneField getField(String name) {
        return fields.get(name);
    }

    public OneMethod getMethod(String name) {
        return methods.get(name);
    }
    
    public List<OneField> getFields() {
        return new ArrayList(fields.values());
    }

    public List<OneMethod> getMethods() {
        return new ArrayList(methods.values());
    }
}

class OneField
{
    public Field field;

    public String name;
    public boolean isStatic;

    public OneField(Field field)
    {
        this.field = field;
        this.name = field.getName();
        this.isStatic = Modifier.isStatic(field.getModifiers());
    }

    public Object getValue(Object obj) throws Exception {
        return field.get(obj);
    }

    public void setValue(Object obj, Object value) throws Exception {
        field.set(obj, value);
    }
}

class OneMethod
{
    public Method method;

    public String name;
    public boolean isStatic;

    public OneMethod(Method method)
    {
        this.method = method;
        this.name = method.getName();
        this.isStatic = Modifier.isStatic(method.getModifiers());
    }

    public Object call(Object obj, List<Object> args) throws Exception {
        return method.invoke(obj, args.toArray(new Object[args.size()]));
    }
}
